### ISSUE TITLE
Entity without relations must not have an empty FilterGroupJoin

### ISSUE THREAD

**Issue by popovserhii** [Ecodev/graphql-doctrine#22]
I'm trying to implement a filter to model but instead of I'm getting an error (in GraphiQL) `Error: MarketplaceFilterGroupJoin fields must be an object with field names as keys or a function which returns such an object`.

I'm doing something wrong? There are some requirements for default filtering?

My $queryType:
```php
$queryType = new ObjectType([
	'name' => 'query',
	'fields' => [
		'marketplaces' => [
			'type' => Type::listOf($types->getOutput(Marketplace::class)),
			'args' => [
				[
					'name' => 'filter',
					'type' => $types->getFilter(Marketplace::class),
				],
				[
					'name' => 'sorting',
					'type' => $types->getSorting(Marketplace::class),
				],
			],
			'resolve' => function ($root, $args) use ($types) {
				$queryBuilder = $types->createFilteredQueryBuilder(Marketplace::class, $args['filter'] ?? [], $args['sorting'] ?? []);
				$result = $queryBuilder->getQuery()->getArrayResult();
				return $result;
			},
		],
	],
]);
```
Realization is getting from [Quick star](https://github.com/Ecodev/graphql-doctrine#quick-start)

**Comment by PowerKiKi** [comment]
There is no requirement for default filtering. It should work out of the box. Your code seems to be correct...

I am not sure when the error happen. When you run a query ? which one ?

You should try validating your schema (in your unit tests) with the `\GraphQL\Type\Schema::assertValid()`

**Comment by popovserhii** [comment]
@PowerKiKi, thanks for a quick response

I've added `\GraphQL\Type\Schema::assertValid()` for assert schema and this is thrown exception `Input Object type MarketplaceFilterGroupJoin must define one or more fields.`

Here is simple setup of GraphQL Server:
```php
$types = new Types($this->entityManager, $this->container);

// Configure default field resolver to be able to use getters
GraphQL::setDefaultFieldResolver(new DefaultFieldResolver());

try {
    $queryType = new ObjectType([
        'name' => 'query',
        'fields' => [
            'marketplaces' => [
                'type' => Type::listOf($types->getOutput(Marketplace::class)), // Use automated ObjectType for output
                'args' => [
                    [
                        'name' => 'filter',
                        'type' => $types->getFilter(Marketplace::class), // Use automated filtering options
                    ],
                    [
                        'name' => 'sorting',
                        'type' => $types->getSorting(Marketplace::class), // Use automated sorting options
                    ],
                ],
                'resolve' => function ($root, $args) use ($types) {
                    $queryBuilder = $types->createFilteredQueryBuilder(Marketplace::class, $args['filter'] ?? [], $args['sorting'] ?? []);
                    $result = $queryBuilder->getQuery()->getArrayResult();
                    return $result;
                },
            ],
        ],
        'resolveField' => function($val, $args, $context, ResolveInfo $info) {
            return $this->{$info->fieldName}($val, $args, $context, $info);
        }
    ]);

    $schema = new Schema([
        'query' => $queryType,
    ]);

    $schema->assertValid();

    $server = new StandardServer([
        'schema' => $schema
    ]);

    $server->handleRequest();
} catch (\Exception $e) {
    StandardServer::send500Error($e);
}
```

Markerplace model
```php
/**
 * @ORM\Entity()
 * @ORM\Table(name="marketplace")
 */
class Marketplace
{
    #use DomainAwareTrait;
    /**
     * @var int
     * @ORM\Id
     * @ORM\GeneratedValue(strategy="AUTO")
     * @ORM\Column(type="integer", options={"unsigned":true})
     */
    private $id;

    /**
     * @var string
     * @ORM\Column(type="string", length=14, unique=true, options={"unsigned":true})
     */
    private $code;

    /**
     * @var string
     * @ORM\Column(type="string", nullable=true, length=255)
     */
    private $name;

    /**
     * @return string
     */
    public function getId()
    {
        return $this->id;
    }

    /**
     * @param string $id
     * @return Marketplace
     */
    public function setId($id): Marketplace
    {
        $this->id = $id;

        return $this;
    }

    /**
     * @return string
     */
    public function getCode(): string
    {
        return $this->code;
    }

    /**
     * @param string $code
     * @return Marketplace
     */
    public function setCode(string $code): Marketplace
    {
        $this->code = $code;

        return $this;
    }

    /**
     * @return string
     */
    public function getName(): string
    {
        return $this->name;
    }

    /**
     * @param string $name
     * @return Marketplace
     */
    public function setName(string $name): Marketplace
    {
        $this->name = $name;

        return $this;
    }
}
```

**Comment by PowerKiKi** [comment]
I see that `Marketplace` has no relation at all with any other entities. This means that there is nothing to be used as a join filter in `MarketplaceFilterGroupJoin`. So `MarketplaceFilterGroupJoin` is effectively empty, but it is invalid to have an "empty" GraphQL type.

Somehow this never happened in our projects, but it clearly is something that should be supported and can be considered a bug. I'll see what I can do...

**Comment by PowerKiKi** [comment]
Please try 65bf449a94ae9e7cb1c82f91d2d5f00cc59e85fd and let me know if it's fixed

**Comment by popovserhii** [comment]
Yes, these changes have fixed my problem. Thanks :).
When do you plan to release these changes?

**Comment by PowerKiKi** [comment]
Released as 5.0.1
